### PR TITLE
Openstack - Include Network Cloud Config for Correct NICs

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

It was observed that although we correctly attached O7k ports to instances based on spaces constraints, the on-machine devices were not available.

This change ensures that Netplan is rendered correctly on multi-NIC OpenStack instances in order that NICs reflect the ports that were added to them.

We previously rendered a nil `CloudConfig` for O7k servers. Now we generate a new one and include network config that reflects the ports that we attach.

## QA steps

Ensure you have a suitable OpenStack for testing multi-NIC support:
https://discourse.jujucharms.com/t/openstack-multi-space-support/2672

After adding a machine with multi-space constraints, run:
```bash
juju run --machine 5 -- ip a
```

The interfaces should be in the up state and reflect the addresses/subnets that we confined the machine to.

## Documentation changes

None

## Bug reference

N/A

### CODE REVIEW

File: provider/openstack/cinder.go [hunk 1]
```diff
@@ -69,9 +69,9 @@ func newCinderConfig(attrs map[string]interface{}) (*cinderConfig, error) {
 }
 
 // StorageProviderTypes implements storage.ProviderRegistry.
-func (env *Environ) StorageProviderTypes() ([]storage.ProviderType, error) {
+func (e *Environ) StorageProviderTypes() ([]storage.ProviderType, error) {
```
Comment (manadart) [inline provider/openstack/cinder.go]: This is just receiver name consistency. File can be ignored in review.

**Comment by manadart** [comment]
[Ensure Openstack multi-NIC machines are reflected in Netplan](https://trello.com/c/lkqJQnE9/2445-ensure-openstack-multi-nic-machines-are-reflected-in-netplan)

File: provider/openstack/provider.go [hunk 2]
```diff
@@ -1459,20 +1467,37 @@ func (e *Environ) networksForInstance(args environs.StartInstanceParams) ([]nova
...
+
+		netInfo[i] = corenetwork.InterfaceInfo{
+			InterfaceName: fmt.Sprintf("eth%d", i),
+			MACAddress:    port.MACAddress,
+			Addresses:     corenetwork.NewProviderAddresses(ips...),
+			ConfigType:    corenetwork.ConfigDHCP,
+		}
 	}
 
+	err = cloudCfg.AddNetworkConfig(netInfo)
```
Comment (SimonRichardson) [inline provider/openstack/provider.go]: Excellent, it's a shame that we don't invert this, so adding a network config isn't magic.

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**